device/quanta/x86_64-quanta_ix1b_rglbmc-r0/plugins/psuutil.py

```diff
@@ -40,12 +42,18 @@ def my_log(txt):
     return
 
 
-def log_os_system(cmd, show):
+def log_os_system(cmd1_args, cmd2_args, show):
+    cmd = ' '.join(cmd1_args) + ' | ' + ' '.join(cmd2_args)
     logging.info('Run :'+cmd)
     status = 1
     output = ""
     try:
-        output = subprocess.check_output(cmd, shell=True, universal_newlines=True)
+        p1 = subprocess.Popen(cmd1_args, universal_newlines=True, stdout=subprocess.PIPE)
+        p2 = subprocess.Popen(cmd2_args, universal_newlines=True, stdin=p1.stdout, stdout=subprocess.PIPE)
+        p1.stdout.close()
+        output = p2.communicate()[0]
+        if p2.returncode == 1:
```

Comment:
>p2.returncode [](http://example.com/codeflow?start=11&length=13)

Should we check both p1 p2 return non zero?

Reply:
p1 will always be None regardless of failure or success.

Reply:
If use `with`, p1 and p2 can be either 0 or 1

Reply:
sorry I mean check p1/p2.returncode is zero or not. I do not mean the stdout.

Reply:
Fixed to check p2.returncode != 0, p1.returncode will always be None

Reply:
> p1.returncode will always be None
This is strange, could you double check?